src/main/java/org/apache/sling/feature/io/file/spi/ArtifactProvider.java

```diff
@@ -52,4 +56,14 @@
      * @return A file if the artifact exists or {@code null}
      */
     File getArtifact(String url, String relativeCachePath);
+
+    default ArtifactHandler getArtifactHandler(String url, String relativeCachePath) {
+        File result = getArtifact(url, relativeCachePath);
+        if (result != null && result.exists()) {
+            return new ArtifactHandler(url, result);
+        }
+        else {
+            return null;
+        }
+    }
 }
```

Comment:
This adds a dependency to the io.file package which so far has been avoided (for circular dependencies between those packages). I think we should avoid this
As this adds new api, the package version needs to be updated
If such a new method (in whatever form) is added, could we maybe deprecate getArtifact() ? This might also need adjustments to the javadoc, as getArtifact might return null while getArtifactHandler might not (if I understand everything correctly)